src/batch-submitter/batch-submitter.ts

```diff
@@ -131,22 +131,34 @@ export abstract class BatchSubmitter {
   }
 
   protected _shouldSubmitBatch(batchSizeInBytes: number): boolean {
+    const currentTimestamp = Date.now()
     const isTimeoutReached =
       this.lastBatchSubmissionTimestamp + this.maxBatchSubmissionTime <=
-      Date.now()
+      currentTimestamp
     if (batchSizeInBytes < this.minTxSize) {
       if (!isTimeoutReached) {
         this.log.info(
           'Skipping batch submission. Batch too small & max submission timeout not reached.',
           {
             batchSizeInBytes,
             minTxSize: this.minTxSize,
+            lastBatchSubmissionTimestamp: this.lastBatchSubmissionTimestamp,
+            currentTimestamp,
           }
         )
         return false
       }
-      this.log.info('Timeout reached.')
+      this.log.info('Timeout reached, proceeding with batch submission.', {
+        batchSizeInBytes,
+        lastBatchSubmissionTimestamp: this.lastBatchSubmissionTimestamp,
+        currentTimestamp,
+      })
     }
+    this.log.info('Sufficient batch size, proceeding with batch submission.', {
```

Comment:
Would change to just `Proceeding with batch submission` because it could be the case that the `timeout` was reached

Reply:
Actually, if you wanna make these logs really good then we should add a `submissionReason` variable at the start and set the reason. And then based on the reason print out these two logs

Reply:
fixed by adding extra `return true`!